### PR TITLE
Convert config to yaml.

### PR DESCRIPTION
**Pull request by devilmonastery**
Adding a yaml config to replace the "go as config lang".

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (shiinee) [inline .vscode/launch.json]: is this something you actually want in the repo?

File: ntp-prometheus.go [hunk 2]
```diff
@@ -140,30 +175,20 @@ func singleprobe(group, target string, conn net.Conn) error {
...
-			if err != nil {
-				log.Printf("error: %s-%s(%s): %v", group, target, hostport, err)
-			}
-		} else {
-			err := singleprobe(group, target, conn)
-			if err != nil {
-				log.Printf("error: %s-%s(%s): %v", group, target, hostport, err)
-			}
+		if err == nil {
+			err = singleprobe(conf, conn)
```
Comment (shiinee) [inline ntp-prometheus.go:183]: log error here?

Comment (shiinee) [inline ntp-prometheus.go:183]:
```suggestion
			err = singleprobe(conf, conn)
			log.Printf("error: %s-%s(%s): %v", conf.Group, conf.Name, conf.Hostport, err)
```

File: ntp-prometheus.go [hunk 3]
```diff
@@ -140,30 +175,20 @@ func singleprobe(group, target string, conn net.Conn) error {
...
 		}
 		elapsed := time.Now().Sub(start)
-		delay := 10.0 - elapsed.Seconds()
-		if delay < 4 {
-			delay = 4
+		delay := float64(conf.Interval) - elapsed.Seconds()
+		if delay < minInterval {
+			delay = minInterval
```
Comment (shiinee) [inline ntp-prometheus.go:193]:
```suggestion
		delay := math.Max(float64(conf.Interval) - elapsed.Seconds(), minInterval)
```

File: ntp-prometheus.go [hunk 4]
```diff
@@ -140,30 +175,20 @@ func singleprobe(group, target string, conn net.Conn) error {
...
 		}
-		if delay > 10 {
-			delay = 10
+		if delay > conf.Interval {
+			delay = conf.Interval
 		}
```
Comment (shiinee) [inline ntp-prometheus.go]:
```suggestion
		delay = math.Min(delay, conf.Interval)
```

File: ntp-prometheus.yaml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  - name: "time2.google.com(ipv4)"
+    group: "google"
+    hostport: "216.239.35.4:123"
+  - name: "time3.google.com(ipv4)"
+    group: "google"
+    hostport: "216.239.35.8:123"
+  - name: "time4.google.com(ipv4)"
+    group: "google"
+    hostport: "216.239.35.12:123"
+  # By IP address, iov6
```
Comment (shiinee) [inline ntp-prometheus.yaml]:
```suggestion
  # By IP address, ipv6
```

**Review by shiinee — CHANGES_REQUESTED** [review]

**Review by devilmonastery — COMMENTED** [review]
Thanks!

**Review by shiinee — APPROVED** [review]

**Pull request merged**